Difi.SikkerDigitalPost.Klient.Tester/SmokeTestsHelper.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Difi.SikkerDigitalPost.Klient.Api;
+using Difi.SikkerDigitalPost.Klient.Domene.Entiteter.Aktører;
+using Difi.SikkerDigitalPost.Klient.Domene.Entiteter.Kvitteringer;
+using Difi.SikkerDigitalPost.Klient.Domene.Entiteter.Kvitteringer.Forretning;
+using Difi.SikkerDigitalPost.Klient.Domene.Entiteter.Kvitteringer.Transport;
+using Difi.SikkerDigitalPost.Klient.Domene.Entiteter.Post;
+using Difi.SikkerDigitalPost.Klient.Domene.Enums;
+using Difi.SikkerDigitalPost.Klient.Tester.Utilities;
+using Difi.SikkerDigitalPost.Klient.XmlValidering;
+using Xunit;
+
+namespace Difi.SikkerDigitalPost.Klient.Tester
+{
+    internal class SmokeTestsHelper
+    {
+        private readonly Miljø _miljø;
+        private SikkerDigitalPostKlient _klient;
+        private Forsendelse _forsendelse;
+        private Transportkvittering _transportkvittering;
+        private Forretningskvittering _forretningskvittering;
+
+        public SmokeTestsHelper(Miljø miljø)
+        {
+            _miljø = miljø;;
+        }
+
+        public SmokeTestsHelper Create_Digital_Forsendelse_with_multiple_documents()
+        {
+            _forsendelse = DomainUtility.GetDigitalDigitalPostWithNotificationMultipleDocumentsAndHigherSecurity();
+
+            return this;
+        }
+
+        public SmokeTestsHelper Create_Physical_Forsendelse()
+        {
+            _forsendelse = DomainUtility.GetFysiskPostSimple();
+
+            return this;
+        }
+
+        public SmokeTestsHelper Create_digital_forsendelse_with_different_sender()
+        {
+            _forsendelse = DomainUtility.GetForsendelseSimple();
+            _forsendelse.Avsender = new Avsender("984661185") {Avsenderidentifikator = "digipost"};
+
+            return this;
+        }
+
+        public SmokeTestsHelper Send()
+        {
+            Assert_state(_forsendelse);
+
+            const string bringOrganisasjonsnummer = "988015814";
+            const string bringThumbprint = "2d7f30dd05d3b7fc7ae5973a73f849083b2040ed";
+
+            _klient = new SikkerDigitalPostKlient(new Databehandler(bringOrganisasjonsnummer, bringThumbprint), new Klientkonfigurasjon(_miljø));
```

Comment:
Hva med å initialisere denne i konstruktøren heller?

Klienten er strengt talt ikke _tilstand_ – og skal vel ikke endre seg i løpet av testen. Det gir heller ikke særlig mening at den opprettes som en del av sendingen. Dersom du ønsker å sende flere forsendelser i samme test burde det ikke være nødvendig å lage en ny klient hver gang.

Added bonus: du slipper å kalle `Assert_state` for å verifisere at klienten er tilstede i f.eks. `Expect_Message_Response_To_Be_TransportOkKvittering`.